server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -0,0 +1,275 @@
+                    : Setting.timeSetting(key, (Setting<TimeValue>) fallback.getSetting(), propertyArray);
+            case ByteSizeValue:
+                return fallback == null
+                    ? Setting.byteSizeSetting(key, (ByteSizeValue) defaultValue, propertyArray)
+                    : Setting.byteSizeSetting(key, (Setting<ByteSizeValue>) fallback.getSetting(), propertyArray);
+            case Version:
+                // No fallback option on this method
+                return Setting.versionSetting(key, (Version) defaultValue, propertyArray);
+            default:
+                // This Should Never Happen (TM)
+                throw new UnsupportedOperationException("A SettingType has been added to the enum and not handled here.");
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        // Write the type
+        out.writeEnum(type);
+        // Write the key
+        out.writeString(setting.getKey());
+        // Write the default value
+        writeDefaultValue(out, setting.getDefault(Settings.EMPTY));
+        // Write a boolean specifying whether the fallback settings is null
+        boolean hasFallback = setting.fallbackSetting != null;
+        out.writeBoolean(hasFallback);
+        if (hasFallback) {
+            new WriteableSetting(setting.fallbackSetting, type).writeTo(out);
+        }
+        // We are using known types so don't need the parser
+        // We are not using validator
+        // Write properties
+        out.writeEnumSet(setting.getProperties());
+    }
+
+    private void writeDefaultValue(StreamOutput out, Object defaultValue) throws IOException {
+        switch (type) {
+            case Boolean:
+                out.writeBoolean((boolean) defaultValue);
+                break;
+            case Integer:
+                out.writeInt((int) defaultValue);
+                break;
+            case Long:
+                out.writeLong((long) defaultValue);
+                break;
+            case Float:
+                out.writeFloat((float) defaultValue);
+                break;
+            case Double:
+                out.writeDouble((double) defaultValue);
+                break;
+            case String:
+                out.writeString((String) defaultValue);
+                break;
+            case TimeValue:
+                TimeValue tv = (TimeValue) defaultValue;
+                out.writeLong(tv.duration());
+                out.writeString(tv.timeUnit().name());
+                break;
+            case ByteSizeValue:
+                ((ByteSizeValue) defaultValue).writeTo(out);
+                break;
+            case Version:
+                Version.writeVersion((Version) defaultValue, out);
+                break;
+            default:
+                // This Should Never Happen (TM)
+                throw new UnsupportedOperationException("A SettingType has been added to the enum and not handled here.");
```

Comment:
`IillegalArgumentException` please, if we ended up with the value we didn't expect that is illegal state